app/controllers/open_ai_controller.rb

```diff
@@ -0,0 +1,50 @@
+class OpenAiController < ApplicationController
+  def show
+    client = OpenAI::Client.new
+
+    @actions = YAML.load(File.read(Rails.root.join("config/arrays/wiki", "actions.yml")))
+    @values = YAML.load(File.read(Rails.root.join("config/arrays/wiki", "values.yml")))
+
+    merged_array = @actions.merge(@values)
+
+    description = ""
+    merged_array.each do |item|
+      if item.is_a?(Array) && item[1].is_a?(Hash) && item[1]['en-US'] == params[:prompt]
+        description = item[1]['description']
+        break
+      end
+    end
+
+    system_prompt = "You are a teacher, trying your best to explain programming terms within the context of the Overwatch Workshop.
+      You will be asked a question about a specific Overwatch Workshop topic, your job is to explain that topic as well as you can.
+      While many of the questions you will be asked are about specific Overwatch Workshop tools, many of them will apply to programming in general.
+      You may assume that the questions are asked by someone with very limited programming knowledge, but with large knowledge of Overwatch itself.
+      Because of this it might be useful to explain something within the context of Overwatch.
+      Please use Markdown to make certain keywords bold or italic."
+
+    prompt = "Explain the #{params[:category]} \"#{params[:prompt]}\".
```

Comment:
I'm sure it doesn't actually harm anyone, but this is using user input on the prompt, meaning one malicious people can inject things into the prompt (whereas it seems the purpose of this prompt is to be somewhat static).
How about adding some checks to see that the Category and Prompt actually match the available categories and wiki page titles on the Wiki?

Reply:
I guess there is already a sort of check like this just to get the description, but if the check fails then `description` is empty, and the request still goes through.

Maybe just fail the request if no description could be found?

Reply:
Good idea! There actually are a few examples of values without descriptions, but the check can be within the same loop